### PR TITLE
Add Churn

### PR DESCRIPTION
**Pull request by ndharasz**
- update scoring function docs
- add churn calculation function
- add prediction cleaning function
- add testing for new functions and top_bottom function

### CODE REVIEW

**Comment by michael-phillips-data** [comment]
where does the prediction cleaning function get used? 
Is it like - all scoring functions are applied to the cleaned predictions always?

**Comment by ndharasz** [comment]
> where does the prediction cleaning function get used?

Right now the function is only used in diagnostics to clean predictions before calculating churn. We use a similar function to clean submissions for use in the Meta Model and scoring. We only rank and fill in preparation for scoring Signals and Crypto submissions. I will refactor some of our backend code to use this open-sourced version.

File: tests/test_scoring.py [hunk 1]
```diff
@@ -201,10 +202,28 @@ def test_neutralize(self):
...
-        df = pd.DataFrame({
-            "target": s,
-            "prediction": reversed(s)
-        })
+        s = [x / 4 for x in range(5)]
+        df = pd.DataFrame({"target": s, "prediction": reversed(s)})
         numerai_corr(df[["prediction"]], df["target"])
         assert pd.Series(s).equals(df["target"]), f"{s} != {list(df['target'].values)}"
+
+    def test_filter_top_bottom(self):
```
Comment (jelford) [inline tests/test_scoring.py:210]: should be a test with `top_bottom=None`

**Review by jelford — APPROVED** [review]
Just one test case you could add otherwise lgtm

**Pull request merged**